webapp/channels/src/components/user_settings/notifications/user_settings_notifications.tsx

```diff
@@ -235,8 +247,10 @@ class NotificationsTab extends React.PureComponent<Props, State> {
         data.email = this.state.enableEmail;
         data.desktop_sound = this.state.desktopSound;
         data.calls_desktop_sound = this.state.callsDesktopSound;
+        data.calls_mobile_sound = this.state.callsMobileSound;
```

Comment:
Instead of having to add each of these here, what if we just changed line 246 above to:
```js
const data = {...props.user.notify_props};
```
so that any unspecified fields are retained? That way, we won't need to make these same changes the next time we add more fields to that object

Reply:
Facepalm moment. :) I really thought that there was an issue with setting defaults that prevented us from doing that. Thanks Harrison!

Reply:
Just blame the existing code for it. It's really easy to fall into existing patterns 😅